packages/driver/src/multi-domain/test_events.ts

```diff
@@ -0,0 +1,7 @@
+import type { SpecBridgeDomainCommunicator } from './communicator'
+
+export const handleTestEvents = (Cypress: Cypress.Cypress, specBridgeCommunicator: SpecBridgeDomainCommunicator) => {
+  specBridgeCommunicator.on('test:before:run', (...args) => {
```

Comment:
I added some specBridgeCommunicator listeners here: 
https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/src/multi-domain/cypress.ts#L79+L88

They are different events(sync:state and window:load) but I think we should align on a pattern for where we should be defining these events, so someone could logically find them later.

Should we:
A. Add them all to the cypress file
B. Add them to this file (and rename it maybe? multi-domain-event-manager??)
C. Add a file for each event type?
D. Something else entirely?